docs/ProgrammingConcerns.md

```diff
@@ -0,0 +1,170 @@
+# Please read this before contributing to FEX-Emu
+FEX-Emu is mostly a regular C++20 application but we have some interesting problems that most applications don't need to contend with.
+There are concerns with dealing with signals and memory allocation that we must be careful with handling.
+
+# Memory allocation routines
+## What is the problem?
+FEX-Emu needs to allocate memory differently than regular applications. This problem happens because FEX runs both 32-bit and 64-bit guest
```

Comment:
This is backwards: You're not starting with the problem itself but with FEX's strategy of addressing it.

A more approachable intro would be "Games expect the full 32-bit address space to be available to them. Since FEX shares a single address space with its emulated game, FEX must ensure to place its internal allocations outside of that address space whenever possible. To this end...".